src/Compilers/Core/Desktop/CommandLine/CommonCommandLineParser.cs

```diff
@@ -349,8 +355,15 @@ internal static string CheckClientArgsForErrors(IEnumerable<string> args)
                         return CodeAnalysisDesktopResources.KeepAliveIsNotAnInteger;
                     }
                 }
+
+                if (string.Equals(arg.TrimEnd(null), shared, StringComparison.OrdinalIgnoreCase))
```

Comment:
The `TrimEnd(null)` call looks out of place.  Why isn't the code which does the argument parsing doing the trim at that point rather than forcing down to the consumer of the arguments?